### ISSUE TITLE
500 Error Unknown Stat...

### ISSUE THREAD

**Issue by scarvell** [joyent/libuv#390]
Using: Express 3.0 (reported issue https://github.com/visionmedia/express/issues/1110#issuecomment-5348254)

500 Error: UNKNOWN, stat 'D:\USERDATA\ScarvelB\Desktop\opeth\public\wqerty<'

If params.id contains < or > in windows, throws that error.

**Comment by bnoordhuis** [comment]
Sanity check, does this script throw an error?

```
require('fs').statSync('D:\\USERDATA\\ScarvelB\\Desktop\\opeth\\public\\wqerty<');
```

**Comment by scarvell** [comment]
Sure does:

```
 Error: UNKNOWN, unknown error 'D:\USERDATA\ScarvelB\Desktop\opeth\public\wqerty<...
```

**Comment by bnoordhuis** [comment]
Okay, thanks.

@piscisaureus: Can you check this out?

**Comment by piscisaureus** [comment]
@bnoordhuis We have to add a mapping for ERROR_BAD_PATHNAME. Mapping it to ENOENT in the stat case is probably fine. However I don't know what to report when open() is attempted with an invalid filename. Apparently POSIX does not specify an error for this case (which makes sense because invalid path names do not exist on unix).

**Comment by bnoordhuis** [comment]
Maybe translate `ERROR_BAD_PATHNAME` to `ERROR_FILE_NOT_FOUND` or `ERROR_PATH_NOT_FOUND`? Like you say, there is no concept of an invalid path on Unices so uv-win is allowed to cheat here.

**Comment by piscisaureus** [comment]
@bnoordhuis Well I could just translate it to UV_ENOENT.

But the alternative would be to add UV_EBADNAME or UV_ELNAME or something.

**Comment by bnoordhuis** [comment]
> Well I could just translate it to UV_ENOENT.

Right, that's more or less what I mean. UV_EBADNAME would be meaningless on other platforms, let's not add that.

**Comment by piscisaureus** [comment]
Fixed in aea5db5da1483d1ab0a1d3e4f622df312202d027.
